client/components/notice/index.tsx

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, FunctionComponent, useRef } from 'react';
+import classnames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { TimerHandle } from 'calypso/types';
+import Gridicon from 'calypso/components/gridicon';
+import { NoticeStatus, NoticeText } from './types';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+/**
+ * Module constants
+ */
+const GRIDICONS_WITH_DROP = [
+	'add',
+	'cross-circle',
+	'ellipsis-circle',
+	'help',
+	'info',
+	'notice',
+	'pause',
+	'play',
+	'spam',
+];
+
+interface Props {
+	className?: string;
+	duration?: number;
+	onDismissClick: () => void;
+	children: React.ReactNode;
+	status?: NoticeStatus;
+	icon?: string | null;
+	isCompact?: boolean;
+	isLoading?: boolean;
+	showDismiss?: boolean;
+	text?: NoticeText;
```

Comment:
This type can be simply a `React.ReactNode`. It already includes `string` as one of possible values.